lib/ansible/modules/network/dcnm/dcnm_vrf.py

```diff
@@ -0,0 +1,284 @@
+from ansible.module_utils.basic import AnsibleModule
+
+__copyright__ = "Copyright (c) 2019 Cisco and/or its affiliates."
+__license__ = "Cisco Sample Code License, Version 1.0"
+__author__ = "Shrishail Kariyappanavar"
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+DOCUMENTATION = '''
+---
+module: dcnm_vrf
+short_description: Send REST API requests to DCNM controller for VRF operations
+version_added: "2.10"
+description:
+    - "Send REST API requests to DCNM controller for VRF operations - Create, Attach, Deploy"
+options:
+  action:
+    description:
+    - 'Intended VRF operation - Create, Attach or Deploy'
+    choices: ['Create', 'Attach', 'Deploy']
+    required: yes
+    type: str
+  fabric:
+    description:
+    - 'Name of the target fabric for VRF operations'
+    type: str
+    required: yes
+  vrf_name:
+    description:
+    - 'Name of the VRF being created'
+    type: str
+    required: yes
+  vrf_template:
+    description:
+    - 'Name of the config template to be used'
+    type: str
+    required: no
+    default: Default_VRF_Universal
+  vrf_extension_template:
+    description:
+    - 'Name of the extension config template to be used'
+    type: str
+    required: no
+    default: Default_VRF_Extension_Universal
+  vrf_template_config:
+    description:
+    - 'Any additional configs to be supplied'
+    type: str
+    required: no
+  vrf_id:
+    description:
+    - 'Unique ID for the VRF'
+    type: int
+    required: yes
+  serial_numbers_vlans:
+    description:
+    - 'Serial number of the target switch for the VRF and VLAN ID to be used'
+    type: list
+    required: yes
+  deployment:
+    description:
+    - 'Deploy after attach or just attach'
+    type: bool
+    required: no
+    default: False
+author:
+    -  Shrishail Kariyappanavar(@nkshrishail)
+'''
+
+EXAMPLES = '''
+- name: Create VRF
+  dcnm_vrf:
+    action: create
+    fabric: vxlan-fabric
+    vrf_name: ansible-vrf
+    vrf_template: Default_VRF_Universal
+    vrf_extension_template: Default_VRF_Extension_Universal
+    vrf_template_config: ""
+    vrf_id: 9000010
+
+- name: Attach VRF to single switch
+  dcnm_vrf:
+    action: attach
+    fabric: vxlan-fabric
+    vrf_name: ansible-vrf
+    serial_numbers_vlans:
+      - XXXXYYYY: 103
+    deployment: False
+
+- name: Attach VRF to multiple switches
+  dcnm_vrf:
+    action: attach
+    fabric: vxlan-fabric
+    vrf_name: ansible-vrf
+    serial_numbers_vlans:
+      - XXXXYYYY: 103
+      - YYYYXXXX: 104
+      - ZZZXXXYY: 103
+    deployment: False
+
+- name: Deploy VRF
+  dcnm_vrf:
+    action: deploy
+    fabric: vxlan-fabric
+    vrf_name: ansible-vrf
+'''
+
+import datetime
+def logit(msg):
+    with open('/tmp/alog.txt', 'a') as of:
+        d = datetime.datetime.now().replace(microsecond=0).isoformat()
+        of.write("---- %s ----\n%s\n" % (d,msg))
+
+
+def check_vrf_exists(module, conn):
```

Comment:
Seems like this method should have a default return value in the event that response is neither a `dict` or `list`?  Probably return `False`?

Reply:
Yeah, right. Will return False